contracts/cw721-sellable/src/execute.rs

```diff
@@ -101,6 +100,54 @@ pub fn try_list(
     Ok(Response::new().add_attribute("method", "list"))
 }
 
+pub fn try_redeem(
+    deps: DepsMut,
+    info: MessageInfo,
+    address: String,
+    ticket_id: &String,
+) -> Result<Response, ContractError> {
+    let contract = Cw721SellableContract::default();
+
+    // Validate only contract owner can call method
+    let minter = contract.minter.load(deps.storage)?;
+    if info.sender != minter {
+        return Err(ContractError::Unauthorized {});
+    }
+
+    // Load ticket, error if ticket does not exist
+    let mut ticket = contract.tokens.load(deps.storage, ticket_id)?;
+
+    // Make sure owner param matches ticket owner
+    if ticket.owner != address {
+        return Err(ContractError::Unauthorized);
+    }
+
+    // Make sure ticket isn't locked or redeemed
+    match ticket.extension {
```

Comment:
Since you're not exhaustively matching and returning something in every branch, easiest way to do this in rust is with `if let` e.g.

```rust
if let Some(ref metadata) = ticket.extension {
  if metadata.redeemed == Some(true) {
    // one error
  } else if metadata.locked == Some(true) {
    // other error
  }
}
```

Reply:
This eliminates the need to return a unit in the default branch